sljit_src/sljitNativeARM_64.c

```diff
@@ -2034,15 +2035,19 @@ SLJIT_API_FUNC_ATTRIBUTE struct sljit_put_label* sljit_emit_put_label(struct slj
 SLJIT_API_FUNC_ATTRIBUTE void sljit_set_jump_addr(sljit_uw addr, sljit_uw new_target, sljit_sw executable_offset)
 {
 	sljit_ins* inst = (sljit_ins*)addr;
+	SLJIT_UPDATE_WX_FLAGS(inst, inst + 4, 0);
 	modify_imm64_const(inst, new_target);
+	SLJIT_UPDATE_WX_FLAGS(inst, inst + 4, 1);
 	inst = (sljit_ins *)SLJIT_ADD_EXEC_OFFSET(inst, executable_offset);
 	SLJIT_CACHE_FLUSH(inst, inst + 4);
 }
 
 SLJIT_API_FUNC_ATTRIBUTE void sljit_set_const(sljit_uw addr, sljit_sw new_constant, sljit_sw executable_offset)
 {
 	sljit_ins* inst = (sljit_ins*)addr;
+	SLJIT_UPDATE_WX_FLAGS(inst, inst + 4, 0);
 	modify_imm64_const(inst, new_constant);
```

Comment:
why not do the RX <-> RX swap once inside modify_imm64_const() instead?

Reply:
It looks like the whole function is a duplication. This could be size reduction optimization.